src/domain/email/email.repository.ts

```diff
@@ -147,7 +149,11 @@ export class EmailRepository implements IEmailRepository {
     safeAddress: string;
     account: string;
   }): Promise<void> {
-    return this.emailDataSource.deleteEmail(args);
+    const email = await this.emailDataSource.getEmail(args);
+    await this.emailApi.deleteEmailAddress({
+      emailAddress: email.emailAddress.value,
+    });
```

Comment:
The errors that can be returned when deleting the email are either non-existent or not-documented – https://docs.pushwoosh.com/platform-docs/api-reference/email-api#deleteemail

This is something that we need to further clarify as if we get an error that an email was already deleted on the provider side then we should not throw an error, and continue with the deletion on our side instead.